File: plugins/ping/ping.go
// Package ping مثال توضيحي بسيط لا علاقة له بأي ميزة من ميزات بايثون
// الأصلية — الغرض الوحيد منه إثبات أن إضافة خدمة جديدة كاملة لهذا
// المشروع تحتاج فقط: (١) حزمة جديدة كهذه تلتزم بعقد plugins.Service،
// و(٢) سطر واحد يضيفها لقائمة services() في main.go. لا registry، لا
// init()، لا أي تعديل آخر في أي مكان — هذا هو الاختبار العملي لبساطة
// البنية بعد الترحيل.
package ping

import (
	"net/http"

	"sunkenbot/internal/httpx"
	"sunkenbot/internal/plugins"
)

const Description = "مثال توضيحي: خدمة صحة بسيطة تثبت سهولة إضافة خدمة جديدة"

type Service struct{}

func New() *Service { return &Service{} }

func (s *Service) Name() string { return "ping" }

func (s *Service) Routes() []plugins.Route {
	return []plugins.Route{
		{Method: "GET", Pattern: "/ping", Handler: httpx.Handle(s.handlePing)},
	}
}

type pingResult struct {
	Message string `json:"message"`
}

// handlePing يستخدم httpx.Handle مباشرة (لا WrapJSON) لأن /ping بلا جسم
// طلب أصلاً — تماماً كما توثّق internal/httpx: Handle لأي endpoint بلا
// جسم، وWrapJSON للجديد فقط عند وجود جسم JSON.
func (s *Service) handlePing(r *http.Request) (pingResult, error) {
	return pingResult{Message: "pong"}, nil
}

File: python-services/plugins/ping.py
"""
plugins/ping.py — أبسط plugin ممكن: GET /ping/echo

يُظهر كيف تبدو بنية plugin جديد من الصفر:
  1. import Plugin و Route من internal.plugin
  2. كتابة handler عادي
  3. تعريف class يرث Plugin
  4. متغير `plugin` في آخر الملف

هذا كل شيء — لا تعديل في main.py ولا في أي ملف آخر.
"""
from internal.plugin import Plugin, Route


async def _handle_echo() -> dict:
    return {"echo": "pong", "service": "sunkenbot-python"}


class PingPlugin(Plugin):
    description = "فحص بسيط للاتصال"
    requirements = []  # لا يحتاج حزم إضافية

    def name(self) -> str:
        return "ping"

    def routes(self) -> list[Route]:
        return [Route(method="GET", path="/ping/echo", handler=_handle_echo)]


plugin = PingPlugin()
